### PR TITLE
Make the statusMessage option in the StatusChangeHandler Configure the fabric8.mode for the fabric8 plugin.

### PR DESCRIPTION
**Pull request by chirino**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -117,6 +117,7 @@
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
     <maven.compiler.source>1.8</maven.compiler.source>
     <maven.compiler.target>1.8</maven.compiler.target>
+    <fabric8.mode>kubernetes</fabric8.mode>
```
Comment (rhuss) [inline pom.xml]: I'd prefer to add it directly to the f-m-p configuration with 

```xml
<configuration>
    <mode>kubernetes</mode>
    ...
</configuration>
```

It's close where it is used. Can still be overwritten from the outside. 

Plus I'm not 100% sure whether it breaks something different, but I _think_ that the CI build use source s2i (so not f-m-p at all), but not 100% sure. If it used binary s2i then setting this mode would prevent it. @iocanel is my assumption correct and is it save to nail the mode down to Kubernetes?

File: controllers/src/main/java/com/redhat/ipaas/controllers/integration/StatusChangeHandlerProvider.java [hunk 2]
```diff
@@ -37,26 +37,30 @@
 
         class StatusUpdate {
             private Optional<Integration.Status> newStatus;
-            private String message;
+            private Optional<String> message;
```
Comment (rhuss) [inline controllers/src/main/java/com/redhat/ipaas/controllers/integration/StatusChangeHandlerProvider.java]: wouldn't it makes sense to move this property to `errorMessage` as it is then clearer what it is used for (at least at the moment) ?

File: controllers/src/main/java/com/redhat/ipaas/controllers/integration/StatusChangeHandlerProvider.java [hunk 3]
```diff
@@ -37,26 +37,30 @@
...
-                this(status, null);
+                this(Optional.of(status), Optional.empty());
             }
 
             public Optional<Integration.Status> getNewStatus() {
                 return newStatus;
             }
 
-            public String getStatusMessage() {
+            public Optional<String> getStatusMessage() {
```
Comment (rhuss) [inline controllers/src/main/java/com/redhat/ipaas/controllers/integration/StatusChangeHandlerProvider.java]: Don't you have to adapt the usage of this method, too ?

File: pom.xml (hunk 1 above)
Comment (chirino) [inline pom.xml]: ok took this bit out of this PR.  I'll let you figure out how to best do this.

File: controllers/src/main/java/com/redhat/ipaas/controllers/integration/StatusChangeHandlerProvider.java (hunk 2 above)
Comment (chirino) [inline controllers/src/main/java/com/redhat/ipaas/controllers/integration/StatusChangeHandlerProvider.java]: Renamed to mach the fields in the integration.

File: controllers/src/main/java/com/redhat/ipaas/controllers/integration/StatusChangeHandlerProvider.java (hunk 3 above)
Comment (chirino) [inline controllers/src/main/java/com/redhat/ipaas/controllers/integration/StatusChangeHandlerProvider.java]: No, don't think so.

Comment (rhuss) [inline controllers/src/main/java/com/redhat/ipaas/controllers/integration/StatusChangeHandlerProvider.java]: ok, it's fed into an immutable builder which has overloaded methods for this.

File: controllers/src/main/java/com/redhat/ipaas/controllers/integration/StatusChangeHandlerProvider.java (hunk 2 above)
Comment (rhuss) [inline controllers/src/main/java/com/redhat/ipaas/controllers/integration/StatusChangeHandlerProvider.java]: Actually I don't meant the status object, think `newStatus` is  a good fit for a `StatusUpdate`. I meant to rename `message` --> `errorMessage` as it is supposed only be set in the error case.

Comment (rhuss) [inline controllers/src/main/java/com/redhat/ipaas/controllers/integration/StatusChangeHandlerProvider.java]: But 'status' is ok, too (just not so nice ;-)

**Pull request merged**